malaria-app-android/src/main/java/com/peacecorps/malaria/activities/RapidFireGame.java

```diff
@@ -236,17 +241,31 @@ public RapidFireTimeCounter(long millisInFuture, long countDownInterval) {
         @Override
         public void onTick(long l) {
             timer.setText(""+l/1000);
+            millisLeft=l;
         }
 
         @Override
         public void onFinish() {
             counter.cancel();
             prepNextQues();
         }
+    }
 
-
+    @Override
+    protected void onPause() {
+        counter.cancel();
+        counter= null;
+        super.onPause();
     }
 
+    @Override
+    protected void onResume() {
+        if(counter==null){
+            counter= new RapidFireTimeCounter(millisLeft,1000);
```

Comment:
Also @chhavip  I have provoded the null check in onResume() because when game starts for first time , both onCreate() and onResume() are called before onStart() of activity => If I don't provide null check, it will create 2 timers - one in onCreate() and other in onResume()                        
also                        
in onPause() I have to set timer as null because in onResume() I am provoding null check, if I won't set null it won't execute the code in onResume()

as in the mentioned link: http://stackoverflow.com/questions/15443219/how-to-pause-and-resume-a-timertask-timer?rq=1